src/include/win.h

```diff
@@ -315,7 +315,11 @@ extern int	get_dcinfo(char *net_name, char domain_name[], char domain_ctrl[]);
 extern SID*	create_everyone_sid(void);
 extern HANDLE LogonUserNoPass(char *username);
 extern BOOL impersonate_user(HANDLE hlogintoken);
-extern BOOL revert_impersonated_user();
+
+/* wrap revert_impersonated_user to log in case of error */
+#define revert_impersonated_user() _log_wrap_revert_impersonated_user(__func__, __LINE__)
+extern BOOL _log_wrap_revert_impersonated_user(LPCSTR, INT);
```

Comment:
Instead of creating another wrapper of a non-existent function, why don't we modify the existing ?
Noticed later revert_impersonated_user() removed

Reply:
Only the definition header is renamed, so as to be helpful with macro replacement, the invocation will still be called as `revert_impersonated_user`. This is to get the function name and line location of invocation of `revert_impersonated_user` for error logging. The `_log_wrap_` is as per convention followed for wrapping `CloseHandle` and `LocalFree` in win.h ,  hit +1 if you agree.

Reply:
Actually both way we can handle this either rename the definition header or modify only the error logging inside the existing function. Up to you

Reply:
the problem with keeping the same name in the definition header is, due to the introduced macro, we will get compiler error, to solve it we will have to end up doing the ugly `#undef` before the function and then again re define the macro after the function, so I thought its better to keep it with different name. +1 if you agree